packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -581,6 +581,26 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         return this.preferences['terminal.enablePaste'];
     }
 
+    protected get shellPath(): string | undefined {
+        if (isWindows) {
```

Comment:
that's browser OS not backend OS, you can be on windows, but terminal on backend running on linux

I am not sure whether we have an API to fetch backend OS already.

Reply:
or maybe we can push configuration to the backend and resolve it in `create` method?

Reply:
Maybe create an Object with all properties and pass that to the create function, which will decide what to use.
```
const shellPreferences = {
  "Windows": this.preferences['terminal.integrated.shell.windows']
  "OSX": this.preferences['terminal.integrated.shell.osx']
  "Linux": this.preferences['terminal.integrated.shell.linux']
}
```
and then use something like:
```
options.shellPreferences[os.type()]
```
 and pass the result on as `shell`-option to the `terminalFactory`

This would pick the right os based on the backend.

Reply:
yes, I thought about something like that too, can you do?

Reply:
Done.